display/man.rs

```diff
@@ -0,0 +1,258 @@
+use plib::PROJECT_NAME;
+use std::error::Error;
+use std::fmt::Display;
+use std::fs::File;
+use std::io::{self, BufReader, Read};
+use std::path::PathBuf;
+use std::process::{Command, Output};
+use tempfile::NamedTempFile;
+
+#[cfg(target_os = "macos")]
+const MAN_PATH: &str = "/usr/local/share/man";
+
+#[cfg(all(target_family = "unix", not(target_os = "macos")))]
+const MAN_PATH: &str = "/usr/share/man";
+
+/// man - display system documentation
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about)]
+struct Args {
+    /// Interpret name operands as keywords for searching the summary database.
+    #[arg(short)]
+    keyword: bool,
+
+    /// Names of the utilities or keywords to display documentation for.
+    names: Vec<String>,
+}
+
+#[derive(Debug)]
+struct ManError(String);
+
+impl Display for ManError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "man: {}", self.0)
+    }
+}
+
+impl Error for ManError {}
+
+/// Gets name of pager to be used from [PAGER], or default [more] pager.
+///
+/// # Returns
+///
+/// [String] value of pager to be used.
+fn get_pager() -> String {
+    std::env::var("PAGER").unwrap_or("more".to_string())
+}
+
+/// Writes content from buffer reader to temporary file.
+///
+/// # Arguments
+///
+/// `reader` - [BufReader] reader that will used as resource for temporary file.
+///
+/// # Returns
+///
+/// [NamedTempFile] temporary file.
+fn write_to_tmp_file<R: Read>(mut reader: BufReader<R>) -> NamedTempFile {
+    let mut temp_file = NamedTempFile::new().expect("failed to create temp file");
+    io::copy(&mut reader, &mut temp_file).expect("failed to write to temp stdin file");
+    temp_file
+}
+
+/// Gets manpage content from plain file or .gz archieve.
+///
+/// # Arguments
+///
+/// `name` - [str] name of necessary system documentation.
+///
+/// # Returns
+///
+/// Tuple of [NamedTempFile] temporary file with documentation content and section number.
+///
+/// # Errors
+///
+/// Returns [std::io::Error] if file not found or reading to [String] failed.
+fn get_map_page(name: &str) -> Result<(NamedTempFile, i32), io::Error> {
+    let (man_page_path, section) = (1..=9)
+        .flat_map(|section| {
+            let plain_path = format!("{MAN_PATH}/man{section}/{name}.{section}");
+            let gz_path = format!("{plain_path}.gz");
+            vec![(gz_path, section), (plain_path, section)]
+        })
+        .find(|(path, _)| PathBuf::from(path).exists())
+        .ok_or_else(|| io::Error::new(io::ErrorKind::NotFound, "man page not found"))?;
+
+    let source: Box<dyn Read> = if man_page_path.ends_with(".gz") {
+        let file = File::open(man_page_path)?;
+        Box::new(GzDecoder::new(file))
+    } else {
+        Box::new(File::open(man_page_path)?)
+    };
+
+    let reader = BufReader::new(source);
+    let tmp_file = write_to_tmp_file(reader);
+
+    Ok((tmp_file, section))
```

Comment:
Don't create a temp-file. Return the reader instead. Later on the groff util can be piped and reader written into stdin.